src/main.cpp

```diff
@@ -950,7 +950,7 @@ int main(int argc, char* argv[])
     }
   };
   // Start with all the basic parsing steps.
-  for (auto& pass : ast::AllParsePasses(std::move(flags))) {
+  for (auto& pass : ast::AllParsePasses(std::move(flags), args.listing)) {
```

Comment:
The use of `listing` is pretty confusing, because it's an entirely different op. I'd love for us to fix these abstractions and be able to list without the parsers.

As evidence of this confusion, the correct/previous behavior here is actually to pass false (which aligns with eliminating this parameter from AllParsePasses), because otherwise the semantics of parsing are changed and you can actually pass `iter:*` and match with full expansion against all iterators (:cry:).